addons/escoria-core/game/core-scripts/esc/esc_object_manager.gd

```diff
@@ -207,6 +214,10 @@ func register_object(object: ESCObject, room: ESCRoom = null, force: bool = fals
 
 	var objects: Dictionary = _get_room_objects_objects(room_key)
 	objects[object.global_id] = object
+	
+	# If object state is not STATE_DEFAULT, save it in manager's object states 
+	if object.state != ESCObject.STATE_DEFAULT:
+		objects_states[object.global_id] = object.state
```

Comment:
In the future, wouldn't this prevent us from being able to load multiple rooms (for whatever reason, e.g. pre-loading) that contain the same object at the same time? Would it make sense to include these states as part of the larger "room->object" structure for the object manager? Or is this just not a concern?